### PR TITLE
[SSP-2842]Add service class for sending email

### PR DESCRIPTION
**Pull request by bzwei**
https://issues.redhat.com/browse/SSP-2842
https://issues.redhat.com/browse/SSP-2799
https://issues.redhat.com/browse/SSP-2610

Read the html email template and replace with runtime parameters. Fetch members of an approver group. Send customized email per approver.

Initially use the email template from the cloud version. Will need some modifications in a future PR.
Will need another PR to handle certificates for SSL connection to SMTP server.
Will need to decide what to send for plain email.

### CODE REVIEW

File: pinakes/data/email_template.html [hunk 1]
```diff
@@ -0,0 +1,590 @@
...
+              </tr>
+              <!-- end body section -->
+
+              <!-- Footer -->
+              <tr>
+                <td class="rh-footer">
+                  <table role="presentation" border="0" align="center" cellpadding="0" cellspacing="0" width="100%">
+                    <tr>
+                      <td class="rh-footer__text">
+                        This email was sent by Red Hat Ansible Automation Platform
```
Comment (eclarizio) [inline pinakes/data/email_template.html:569]: I'm not sure what language to use here, but we probably don't want to say "Red Hat Ansible Automation Platform". Maybe just "This email was sent by the Pinakes project"?

File: pinakes/data/email_template.html [hunk 2]
```diff
@@ -0,0 +1,590 @@
...
+
+              <!-- Head -->
+              <tr>
+                <td class="rh-head">
+                  <table class="rh-masthead" role="presentation" align="center" border="0" cellpadding="0" cellspacing="0" width="100%">
+                    <tr>
+                      <!--Red Hat logo-->
+                      <td align="center" class="rh-masthead__brand" bgcolor="#151515">
+                        <a href="https://cloud.redhat.com/" target="_blank" class="rh-masthead__brand-link">
+                          <img src="https://$web_url/ui/catalog/fonts/logo_small.svg" alt="Pinakes logo" width="270" height="82" class="rh-masthead__brand-img" />
```
Comment (eclarizio) [inline pinakes/data/email_template.html:457]: Probably don't want to link to the cloud.redhat.com here, and I'm not sure if that logo URL is correct.

File: pinakes/data/email_template.html [hunk 3]
```diff
@@ -0,0 +1,590 @@
...
+                  <table class="rh-content" role="presentation" align="center" border="0" cellpadding="0" cellspacing="0" width="100%">
+
+                    <!-- Content block -->
+                    <tr>
+                      <td class="rh-content__block">
+                        <p>
+                          <strong>$approver_name,</strong>
+                          <br/>
+                          <br/>
+                          The following Automation Services Catalog order requires your approval:
```
Comment (eclarizio) [inline pinakes/data/email_template.html:483]:
```suggestion
                          The following Pinakes order requires your approval:
```

File: pinakes/data/email_template.html [hunk 4]
```diff
@@ -0,0 +1,590 @@
...
+                                <span>
+                                  Process request
+                                </span>
+                              </a>
+                            </td>
+                          </tr>
+                        </table>
+                      </td>
+                    </tr>
+                    <!-- end content block -->
```
Comment (eclarizio) [inline pinakes/data/email_template.html:555]: This approval link doesn't exist yet, right? Maybe we should remove this for now.

**Review by eclarizio — COMMENTED** [review]
I know it's not finalized, but that `email_template.html` has a lot of stuff in it that definitely needs to change, at least for the time being, I think.

Eventually we can probably do a makeover on the whole template, and/or maybe put it in the overrides folder and use it from there so people can make their own template.

**Comment by bzwei** [comment]
@eclarizio We will need more enhancements for the html template. Will be done in another PR.
This PR needs refactoring after #459 is merged.

**Comment by eclarizio** [comment]
> @eclarizio We will need more enhancements for the html template. Will be done in another PR. This PR needs refactoring after #459 is merged.

#459 has been merged so we can rebase/refactor now. Doing the html template changes in another PR should be fine I think, but we should be aware that this current email template is very much a placeholder.

**Review by eclarizio — APPROVED** [review]

**Review by hsong-rh — APPROVED** [review]

**Pull request merged**